web-ng/src/app/components/form-field-editor/form-field-editor.component.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  EventEmitter,
+  OnChanges,
+} from '@angular/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { FieldType } from '../../shared/models/form/field.model';
+
+export interface FieldTypeOptionModel {
+  icon: string;
+  label: string;
+  type: FieldType;
+}
+
+@Component({
+  selector: 'app-form-field-editor',
+  templateUrl: './form-field-editor.component.html',
+  styleUrls: ['./form-field-editor.component.css'],
+})
+export class FormFieldEditorComponent implements OnInit {
+  @Input() label?: string;
+  @Input() required?: boolean;
+  @Input() type?: string;
+  @Output() update = new EventEmitter();
+  @Output() delete = new EventEmitter();
+  fieldTypeOptions: FieldTypeOptionModel[] = [
```

Comment:
I started writing this:
> `FieldTypeOptionModel` is a bit hard to grok at first glance. Can we call it something like `FieldType`, and then the property could be "availableFieldTypes"?

But then I realized, why do we need this at all? Since it's constant, can't it be defined directly in the template?

Reply:
I am using it for mat-select-trigger which requires the element to be index of array type

Reply:
An array is fine here, I was referring to the type and variable names. Option and Model are overloaded terms, and using them here arguably makes things less clear.